DQM/SiStripMonitorClient/python/SiStripClientConfigP5_cff.py

```diff
@@ -36,7 +36,7 @@
     ),
 
 # Parameters for On Demand Monitoring   
-    MonitorSiStripBackPlaneCorrection = OnDemandMonitoring.MonitorSiStripBackPlaneCorrection,
+    MonitorSiStripBackPlaneCorrection = cms.bool(OnDemandMonitoring.MonitorSiStripBackPlaneCorrection),
```

Comment:
@makortel  I need help with this one too: this does not work and crashes with 
ValueError: cms.bool(False) is not a valid <class 'FWCore.ParameterSet.Types.bool'>  (???)
I don't understand why, I tried using eval but it does not work either.

Reply:
The problem is that `OnDemandMonitoring.MonitorSiStripBackPlaneCorrection` is a  `cms.bool()` itself, and `cms.bool(cms.bool(False))` indeed does not work. The following would work
```suggestion
    MonitorSiStripBackPlaneCorrection = cms.bool(OnDemandMonitoring.MonitorSiStripBackPlaneCorrection.value()),
```

Note that below lines like
```py
    MonitorSiStripPedestal      = OnDemandMonitoring.MonitorSiStripPedestal,
```
make `SiStripAnalyser.MonitorSiStripPedestal` and `OnDemandMonitoring.MonitorSiStripPedestal` be the same object, so if someone changes `SiStripAnalyser.MonitorSiStripPedestal` the `OnDemandMonitoring.MonitorSiStripPedestal` changes as well (since they are the same). We typically discourage these kind of "long distance interactions" because they can cause very confusing situations to debug.

One way to reduce boilerplate would be to define a PSet for the options whose names and values are the same between `SiStripAnalyser` and `OnDemandMonitoring`: move those to a PSet that is then used to initialize both EDModule configurations
```
common = cms.PSet(...)
SiStripAnalyser = cms.EDAnalyzer("SiStripAnalyser", common, ...)
OnDemandMonitoring = cms.EDAnalyzer("SiStripMonitorCondDataOnDemandExample", common, ...)
```
The parameters from the `common` PSet are copied to both EDAnalyzers.

Reply:
Thanks!
About the common PSet, since SiStripAnalyser and OnDemandMonitoring are defined in different cfi files, I am sorry for my naive question but, I am not sure how to handle it then

Reply:
You can define the "common PSet" (btw, please use something more descriptive than my `common`) e.g. in a new cfi file that gets imported in `SiStripClientConfigP5_cff` and `OnDemandMonitoring_cfi', or just define it in `OnDemandMonitoring_cfi' since that one is already imported into `SiStripClientConfigP5_cff`.

Reply:
Thanks @makortel 
I tried to implement your suggestion but I have the impression that, unless mistake from my side, the code gets more complicated due to the number of changes involved....